opslevel/datasource_opslevel_team.go

```diff
@@ -1,85 +1,147 @@
-// 			},
-// 			"parent_id": {
-// 				Type:        schema.TypeString,
-// 				Description: "The id of the parent team.",
-// 				Computed:    true,
-// 			},
-// 		},
-// 	}
-// }
-
-// func datasourceTeamRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	resource, err := findTeam("alias", "id", d, client)
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	d.SetId(string(resource.Id))
-// 	d.Set("alias", resource.Alias)
-// 	d.Set("name", resource.Name)
-
-// 	if err := d.Set("members", mapMembershipsArray(resource.Memberships)); err != nil {
-// 		return err
-// 	}
-
-// 	if err := d.Set("parent_alias", resource.ParentTeam.Alias); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("parent_id", resource.ParentTeam.Id); err != nil {
-// 		return err
-// 	}
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure TeamDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &TeamDataSource{}
+
+func NewTeamDataSource() datasource.DataSource {
+	return &TeamDataSource{}
+}
+
+// TeamDataSource manages a Team data source.
+type TeamDataSource struct {
+	CommonDataSourceClient
+}
+
+// TeamDataSourceModel describes the data source data model.
+type TeamDataSourceModel struct {
+	Alias       types.String `tfsdk:"alias"`
+	Id          types.String `tfsdk:"id"`
+	Members     types.List   `tfsdk:"members"`
+	Name        types.String `tfsdk:"name"`
+	ParentAlias types.String `tfsdk:"parent_alias"`
+	ParentId    types.String `tfsdk:"parent_id"`
+}
+
+var memberObjectType = types.ObjectType{
+	AttrTypes: map[string]attr.Type{
+		"email": types.StringType,
+		"role":  types.StringType,
+	},
+}
+
+func membershipsToListValue(ctx context.Context, members *opslevel.TeamMembershipConnection) basetypes.ListValue {
+	if members == nil {
+		return basetypes.NewListNull(memberObjectType)
+	}
+
+	output := make([]attr.Value, len(members.Nodes))
+	for i, membership := range members.Nodes {
+		object := make(map[string]attr.Value)
+		object["email"] = basetypes.NewStringValue(membership.User.Email)
+		object["role"] = basetypes.NewStringValue(membership.Role)
+		output[i] = types.ObjectValueMust(memberObjectType.AttrTypes, object)
+	}
+	return types.ListValueMust(memberObjectType, output)
+}
+
+func NewTeamDataSourceModel(ctx context.Context, team opslevel.Team) TeamDataSourceModel {
+	return TeamDataSourceModel{
+		Alias:       types.StringValue(team.Alias),
+		Id:          types.StringValue(string(team.Id)),
+		Members:     membershipsToListValue(ctx, team.Memberships),
```

Comment:
```suggestion
		Members:     membershipsToListValue(team.Memberships),
```
After func signature update above